### PR TITLE
Docs (update to v20.07) - Port encryption and use of mTLS for all internal gRPC communications

### PR DESCRIPTION
**Pull request by aaroncarey**
Updates 20.07 docs to note the addition of mTLS and port encryption. This is not a cherry-pick of https://github.com/dgraph-io/dgraph/pull/6893 because in 20.07 we did not split the `tls_dir` option into  `tls_cacert`, `tls_node_key`, and `tls_node_cert`.

Fixes DGRAPH-2692

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7243)

### CODE REVIEW

File: wiki/content/deploy/tls-configuration.md [hunk 1]
```diff
@@ -191,40 +232,99 @@ The server will always **request** Client Authentication.  There are four differ
...
+certificate you've imported in the step above and queries/mutations will
 succeed.
 
 ## Using Curl with Client authentication
 
 When TLS is enabled, `curl` requests to Dgraph will need some specific options to work.  For instance (for an export request):
 
 ```
-curl --silent --cacert ./tls/ca.crt https://localhost:8080/admin/export
+curl --silent https://localhost:8080/admin/export
```
Comment (aman-bansal) [inline wiki/content/deploy/tls-configuration.md:327]: --cacert is flag for curl client. we needed this whenever tls is enabled

File: wiki/content/deploy/tls-configuration.md [hunk 2]
```diff
@@ -121,13 +150,16 @@ Important points:
...
 
-The following configuration options are available for Alpha:
+The following configuration options are available for Dgraph Alpha and Dgraph
+Zero nodes:
 
 * `--tls_dir string` - TLS dir path; this enables TLS connections (usually 'tls').
 * `--tls_use_system_ca` - Include System CA with Dgraph Root CA.
-* `--tls_client_auth string` - TLS client authentication used to validate client connection. See [Client Authentication Options](#client-authentication-options) for details.
+* `--tls_client_auth string` - TLS client authentication used to validate client
+  connections from external ports. To learn more, see
```
Comment (aman-bansal) [inline wiki/content/deploy/tls-configuration.md:161]: tls_internal_port_enabled flag is available in 20.07. This specify whether to connect with internal ports with tls config or not

File: wiki/content/deploy/tls-configuration.md [hunk 3]
```diff
@@ -167,7 +206,7 @@ $ dgraph cert -n localhost -c dgraphuser
 $ dgraph alpha --tls_dir tls --tls_client_auth="REQUIREANDVERIFY"
```
Comment (aman-bansal) [inline wiki/content/deploy/tls-configuration.md:206]: tls_internal_port_enabled=true flag is required to enable internal ports encryption

**Review by aaroncarey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7243)* status: 0 of 2 files reviewed, 3 unresolved discussions (waiting on @aman-bansal and @OmarAyo)

___
*[wiki/content/deploy/tls-configuration.md, line 161 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQEoR_W0Z8rwCUuSUr2-r2-161:-MQIelyb1d-iqw9rXVTF:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/2f622fa4e54805312befef8d36631b12f4b08097/wiki/content/deploy/tls-configuration.md#L161)):*
<details><summary><i>Previously, aman-bansal (aman bansal) wrote…</i></summary><blockquote>

tls_internal_port_enabled flag is available in 20.07. This specify whether to connect with internal ports with tls config or not
</blockquote></details>

Done.
___
*[wiki/content/deploy/tls-configuration.md, line 206 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQEoR_W0Z8rwCUuSUr2-r2-206:-MQIf7KTCOEOtscttPLj:b-896fix) ([raw file](https://github.com/dgraph-io/dgraph/blob/2f622fa4e54805312befef8d36631b12f4b08097/wiki/content/deploy/tls-configuration.md#L206)):*
<details><summary><i>Previously, aman-bansal (aman bansal) wrote…</i></summary><blockquote>

tls_internal_port_enabled=true flag is required to enable internal ports encryption
</blockquote></details>

Done.
___
*[wiki/content/deploy/tls-configuration.md, line 327 at r2](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQEoR_W0Z8rwCUuSUr2-r2-327:-MQIfoAZ5d5ON4uyCl7B:brnmdx6) ([raw file](https://github.com/dgraph-io/dgraph/blob/2f622fa4e54805312befef8d36631b12f4b08097/wiki/content/deploy/tls-configuration.md#L327)):*
<details><summary><i>Previously, aman-bansal (aman bansal) wrote…</i></summary><blockquote>

--cacert is flag for curl client. we needed this whenever tls is enabled
</blockquote></details>

Done. I removed the example that showed using `curl` without providing the `-cacert` flag.

**Review by aaroncarey — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7243)* status: all files reviewed, 3 unresolved discussions (waiting on @aman-bansal and @OmarAyo)

**Review by aman-bansal — APPROVED** [review]

**Review by danielmai — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7243)* status: all files reviewed, 5 unresolved discussions (waiting on @aaroncarey, @aman-bansal, and @OmarAyo)

___
*[wiki/content/deploy/tls-configuration.md, line 12 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQZcOZf7UaoXhg49ZHM:-MQZcOZf7UaoXhg49ZHN:b-mcbcob) ([raw file](https://github.com/dgraph-io/dgraph/blob/504ed1ad28b2b1530419680bd7ba2224a8d25ab6/wiki/content/deploy/tls-configuration.md#L12)):*
> ```
> `gRPC-external-private` 
> ```
1. This shouldn't be a `code-formatted`. It should be **in bold**, like the list in Ports Usage.
2. This needs a link to Ports Usage where "gRPC-external-private" is defined.
___
*[wiki/content/deploy/tls-configuration.md, line 269 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQZbO_P-VQ4ac9yL6Ay:-MQZbO_Q20Pn6YFAUM06:b-km8rpf) ([raw file](https://github.com/dgraph-io/dgraph/blob/504ed1ad28b2b1530419680bd7ba2224a8d25ab6/wiki/content/deploy/tls-configuration.md#L269)):*
> ```
> Mac OS X
> ```
macOS

**Review by aaroncarey — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7243)* status: 0 of 2 files reviewed, 5 unresolved discussions (waiting on @aaroncarey, @aman-bansal, @danielmai, and @OmarAyo)

___
*[wiki/content/deploy/tls-configuration.md, line 12 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQZcOZf7UaoXhg49ZHM:-MQZl8lh8lQuQ20GVbFl:b8ukx9b) ([raw file](https://github.com/dgraph-io/dgraph/blob/504ed1ad28b2b1530419680bd7ba2224a8d25ab6/wiki/content/deploy/tls-configuration.md#L12)):*
<details><summary><i>Previously, danielmai (Daniel Mai) wrote…</i></summary><blockquote>

> ```
> `gRPC-external-private` 
> ```
1. This shouldn't be a `code-formatted`. It should be **in bold**, like the list in Ports Usage.
2. This needs a link to Ports Usage where "gRPC-external-private" is defined.
</blockquote></details>

Done (formatting fixed here and elsewhere). The link to Ports Usage is in the following sentence.
___
*[wiki/content/deploy/tls-configuration.md, line 269 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQZbO_P-VQ4ac9yL6Ay:-MQZlP1eBAL-3RUCYNsA:b-376vyo) ([raw file](https://github.com/dgraph-io/dgraph/blob/504ed1ad28b2b1530419680bd7ba2224a8d25ab6/wiki/content/deploy/tls-configuration.md#L269)):*
<details><summary><i>Previously, danielmai (Daniel Mai) wrote…</i></summary><blockquote>

> ```
> Mac OS X
> ```
macOS
</blockquote></details>

Done. Good catch!

**Comment by danielmai** [comment]
___
*[wiki/content/deploy/tls-configuration.md, line 269 at r3](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQZbO_P-VQ4ac9yL6Ay:-MQZnnU7CoIe_r4Xdq07:b-mury0o) ([raw file](https://github.com/dgraph-io/dgraph/blob/504ed1ad28b2b1530419680bd7ba2224a8d25ab6/wiki/content/deploy/tls-configuration.md#L269)):*
<details><summary><i>Previously, aaroncarey (aaroncarey) wrote…</i></summary><blockquote>

Done. Good catch!
</blockquote></details>

The name is just "macOS", not "macOS X".

**Review by danielmai — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7243)* status: 0 of 2 files reviewed, 5 unresolved discussions (waiting on @aaroncarey, @aman-bansal, and @OmarAyo)

___
*[wiki/content/deploy/tls-configuration.md, line 106 at r4](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-MQZoAhTEpCIzAI-bFxU:-MQZoAhTEpCIzAI-bFxV:b9lfvpt) ([raw file](https://github.com/dgraph-io/dgraph/blob/ecf95153ee075f887426784406e0c58df9f98c81/wiki/content/deploy/tls-configuration.md#L106)):*
> ```Markdown
> ### Certificate inspection
> 
> The command **dgraph cert ls** lists all certificates and keys in the **--dir**
> ```

CLI commands and flags can stay in `code-formatted`.

**Review by danielmai — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r4, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7243#-:-MQZqccP5D2MI_3RAEEq:bnfp4nl)* status: all files reviewed, 5 unresolved discussions (waiting on @aaroncarey, @aman-bansal, and @OmarAyo)

**Pull request merged**